src/formats/git.cpp

```diff
@@ -215,30 +216,63 @@ bool GitCommitLog::parseCommit(RCommit& commit) {
             //this isnt a commit we are parsing, abort
             if(commit.timestamp == 0) return false;
 
+            if(!logf->getNextLine(line)) return false;
```

Comment:
The logfile could be an existing file generated with a previous version of Gource's log file command without the hash. It should still continue to work with the existing logfile, so if this line is a file line (starting with ':'), and they didn't request scaling by file size, it needs to recover from that. If they are using file scaling and there's no hash it could fail to parse at that point, ideally with a descriptive error.